### PR TITLE
(chore) Cache install dependencies step in e2e CI workflow

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
During the process of updating the patient chart, a timeout error occurred in the end-to-end GitHub actions. The objective of this PR is to resolve the underlying issue to ensure proper functionality.

## Related Issue

https://issues.openmrs.org/browse/O3-2175

### CODE REVIEW

**Comment by RandilaP** [comment]
@jayasanka-sack @anjula-sack @denniskigen @Piumal1999 Could you please review this

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
There are some changes to the lock file. Could you please undo those changes?

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -64,7 +72,15 @@ jobs:
         with:
           node-version: 16
 
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v3
```
Comment (anjula-sack) [inline .github/workflows/e2e.yml:77]:
```suggestion
            id: cache
            uses: actions/cache@v3
```

Here as well

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -64,7 +72,15 @@ jobs:
...
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
       - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (anjula-sack) [inline .github/workflows/e2e.yml:83]:
```suggestion
            if: steps.cache.outputs.cache-hit != 'true'
```

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -26,7 +26,15 @@ jobs:
         with:
           node-version: 16
 
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v3
```
[the remark this answers is not in the document]
Comment (RandilaP) [reply .github/workflows/e2e.yml:31]: It shows invalid child statement error

[the remark this answers is not in the document]
Comment (anjula-sack) [reply .github/workflows/e2e.yml:31]: yeah it's correct my bad

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -26,7 +26,15 @@ jobs:
         with:
           node-version: 16
```
Comment (anjula-sack) [inline .github/workflows/e2e.yml:27]: Shall we try changing the node versiom

**Comment by Piumal1999** [comment]
Hi @RandilaP, Could you please resolve the conflicts and update the PR? The cache will work now. Refer to esm-patient-management if needed.

**Comment by anjula-sack** [comment]
@RandilaP Please resolve the conflict here

**Comment by Piumal1999** [comment]
Hi @RandilaP @anjula-sack, 
I have already configured the dependency caching feature within pull request #1305. If this pull request is merged before mine, I will revert the changes made in my pull request.

**Comment by anjula-sack** [comment]
I merged the pull request https://github.com/openmrs/openmrs-esm-patient-chart/pull/1305 this is no longer needed

**Pull request closed without merging**